Add monthly update for January 2022 (#3010)
pipe-cd/pipe: docs/content/en/blog/news/january-2022-update.md

---
date: 2022-01-05
title: "January 2022 update"
linkTitle: "January 2022 update"
weight: 997
description: "Development status update to recap what happened in January"
author: Le Van Nghia ([@nghialv](https://twitter.com/nghialv2607))
---

_Published by the PipeCD dev team every month, this update will provide you with news and updates about the project! Please click [here](/blog/2021/11/01/november-2021-update/) if you want to see the last status update._

#### Happy New Year
---
First of all, PipeCD team would like to wish you all a very Happy New Year. 🥳

> 2020 has been a great year for both PipeCD project and PipeCD team. Many features have been introduced by more than 2000 pull requests from 33 contributors. We would like to thank all those wonderful contributors.

Reviewer: 2021 🤔

Author: Nice catch! 😄

![](/images/january-2022-contributor-list.png)

> Stepping into 2021, PipeCD team looks forward to contributing even more to make PipeCD project better and more useful for many users in our OSS community.

Reviewer: ```suggestion
Stepping into 2022, PipeCD team looks forward to contributing even more to make PipeCD project better and more useful for many users in our OSS community.
```

By [v0.23.0](https://github.com/pipe-cd/pipe/releases/tag/v0.23.0), PipeCD team has brought that feature to reality. It will allow you to roll out applications to multiple clusters/regions gradually or to promote applications across environments. Those deployment strategies can be done simply by specifying a chain of applications in the `postSync` field of the first application.

```yaml
apiVersion: pipecd.dev/v1beta1
kind: KubernetesApp
spec:
  ...
  postSync:
    chain:
      applications:
        - name: second-app
          kind: KUBERNETES
        - name: third-app
          kind: CLOUDRUN
```

Currently, in-chain applications can be specified only by `name`, `kind` but finding applications by `label` will also be supported in the near future as well. Since PipeCD is supporting multiple application kinds (such as Kubernetes, Terraform, CloudRun...) so applications from different kinds are allowed to be included in a chain.

For more details, please check out its [documentation](https://pipecd.dev/docs/user-guide/deployment-chain/) page.

##### Environment is no longer required and will be replaced by Label

> Environment was changed to be optional while registering Piped or Application. You still can use it from the web console or specify the environment for an application via the `envName` field in the application configuration file, but we are planning to completely remove the environment concept in the future. As an alternative, a new `Label` concept has been introduced and `Environment` will be just a special label.

Reviewer: ```suggestion
Environment was changed to be optional while registering Piped or Application. You still can use it from the web console or specify the environment for an application via the `envName` field in the application configuration file, but we are planning to completely remove the environment concept in the near future. As an alternative, a new `Label` concept has been introduced, and `Environment` can be imagined as a particular label.
```
nits

An application can contain one or more labels in its application configuration file as below:

```yaml
apiVersion: pipecd.dev/v1beta1
kind: TerraformApp
spec:
  name: simple-app
  envName: production
  labels:
    #env: production
    service: backend
  ...
```

Filtering application or deployment by the label is being implemeted currently. We hope that it will be shipped in the next release.

> ##### Simplifying the way of registering application

Reviewer: ```suggestion
##### Simplifying the way of registering applications
```

> Before [v0.23.0](https://github.com/pipe-cd/pipe/releases/tag/v0.23.0), while registering a new application user has to configure the application information manually. But now most of the information can be specified in the application configuration file. And Piped agents automatically find the un-registered applications to suggest to users. It means that all you have to do is just a few clicks on the web console.

Reviewer: ```suggestion
Before [v0.23.0](https://github.com/pipe-cd/pipe/releases/tag/v0.23.0), while registering a new application user has to configure the application information manually via a registration form. For now, most of the information can be specified in the application configuration file. And Piped agents automatically find the un-registered applications to suggest to users. It means that all you have to do is just a few clicks on the web console.
```

![](/images/january-2022-piped-connection-status.png)

##### More controls on triggering the deployment

By default, when a new merged pull request touches an application, a new deployment for that application will be triggered to execute the sync process.

From `v0.22.0`, a new [`trigger`](https://pipecd.dev/docs/user-guide/configuration-reference/#deploymenttrigger) field has been added to the application configuration file to allow users to customize the triggering condition. For example, using `onOutOfSync` to enable the ability to attempt to resolve `OUT_OF_SYNC` state whenever a configuration drift has been detected.

The deployment triggering of each application can be configured with the following fields:

* `onCommit`: controls triggering new deployment when newly added Git commits touched the application.
* `onCommand`: controls triggering new deployment when received a new `SYNC` command from the web console or `pipectl`
* `onOutOfSync`: controls triggering new deployment when application is at `OUT_OF_SYNC` state. Enabling this will force Piped to always attempt to keep the application as synced as possible.

> 

Reviewer: should we add `onChain` trigger description here as well, since it's released from `v0.23.0` already?

Author: Nice catch! Let me add it.

Author: Fixed!

---
date: 2022-01-05
title: "January 2022 update"
linkTitle: "January 2022 update"
weight: 997
description: "Development status update to recap what happened in January"
author: Le Van Nghia ([@nghialv](https://twitter.com/nghialv2607))
---

_Published by the PipeCD dev team every month, this update will provide you with news and updates about the project! Please click [here](/blog/2021/11/01/november-2021-update/) if you want to see the last status update._

#### Happy New Year
---
First of all, PipeCD team would like to wish you all a very Happy New Year. 🥳

2021 has been a great year for both PipeCD project and PipeCD team. Many features have been introduced by more than 2000 pull requests from 33 contributors. We would like to thank all those wonderful contributors.

![](/images/january-2022-contributor-list.png)

Stepping into 2022, PipeCD team looks forward to contributing even more to make PipeCD project better and more useful for many users in our OSS community.

> #### What's changed
> ---
> 
> Since the last report, PipeCD team has introduced 2 releases ([v0.22.0](https://github.com/pipe-cd/pipe/releases/tag/v0.22.0), [v0.23.0](https://github.com/pipe-cd/pipe/releases/tag/v0.23.0)). These releases bring many updates to help PipeCD becomes more stable as well as introduce some interesting features. In this blog post, we will recap those new features. For all other changes, please check out each release note to see.
> 
> ##### Deployment chain

Reviewer 2: I'm aware that previous posts were using h4 and h5 for these. But honestly it's a bit unreadable. Especially h5, at a glance, it's the same as the content. What do you find on changing all heading levels?

```suggestion
### What's changed
---

Since the last report, PipeCD team has introduced 2 releases ([v0.22.0](https://github.com/pipe-cd/pipe/releases/tag/v0.22.0), [v0.23.0](https://github.com/pipe-cd/pipe/releases/tag/v0.23.0)). These releases bring many updates to help PipeCD becomes more stable as well as introduce some interesting features. In this blog post, we will recap those new features. For all other changes, please check out each release note to see.

#### Deployment chain
```

Reviewer 2: ### Before
![image](https://user-images.githubusercontent.com/19730728/148499863-d540a5dd-6f52-4bbe-992c-49343b5ae692.png)

### After
![image](https://user-images.githubusercontent.com/19730728/148499805-1ed49f84-3326-4de8-9626-beb02ed1c872.png)

Author: Nice idea. Let me adopt it!

Author: Updated both of them. Thank you.

![](/images/january-2022-piped-connection-status.png)

##### More controls on triggering the deployment

By default, when a new merged pull request touches an application, a new deployment for that application will be triggered to execute the sync process.

From `v0.22.0`, a new [`trigger`](https://pipecd.dev/docs/user-guide/configuration-reference/#deploymenttrigger) field has been added to the application configuration file to allow users to customize the triggering condition. For example, using `onOutOfSync` to enable the ability to attempt to resolve `OUT_OF_SYNC` state whenever a configuration drift has been detected.

The deployment triggering of each application can be configured with the following fields:

* `onCommit`: controls triggering new deployment when newly added Git commits touched the application.
* `onCommand`: controls triggering new deployment when received a new `SYNC` command from the web console or `pipectl`
* `onOutOfSync`: controls triggering new deployment when application is at `OUT_OF_SYNC` state. Enabling this will force Piped to always attempt to keep the application as synced as possible.

> * `onChain`: controls triggering new deployment when the application is counted as a node of a deployment chain.

Reviewer: ```suggestion
* `onChain`: controls triggering new deployment when the application is counted as a node of a deployment chain. (this configuration is added from `v0.23.0`)
```
nits
